src/Avalonia.Base/Compatibility/UnmanagedCallersOnlyAttribute.cs

```diff
@@ -0,0 +1,14 @@
+﻿#pragma warning disable
+#nullable enable
+
+#if !NET6_0_OR_GREATER
+namespace System.Runtime.InteropServices
+{
+    [global::System.AttributeUsage(global::System.AttributeTargets.Method, Inherited = false)]
+    internal sealed class UnmanagedCallersOnlyAttribute : global::System.Attribute
```

Comment:
FYI: I used function pointers in netstandard2.0, and it even works on net framework.

Reply:
IIRC one of the reasons for that attribute was to ensure that the function has a proper native->managed transition and is even callable with the corresponding calling convention.

.NET Framework isn't aware of the attribute and isn't likely to setup a proper native->managed transition.

Reply:
So I believe that it's not safe to do.